lib/backpex/resource.ex

```diff
@@ -265,6 +265,25 @@ defmodule Backpex.Resource do
   * `fields` (list): A list of atoms representing the fields to be selected and potentially preloaded.
   """
   def get!(id, repo, schema, item_query, fields) do
+    query_record(id, repo, schema, item_query, fields, :one!)
+  end
+
+  @doc """
+  Gets a database record with the given fields by the given id. Returns nil if no result was found.
+
+  ## Parameters
+
+  * `id`: The identifier for the specific item to be fetched.
+  * `repo` (module): The repository module.
+  * `schema`: The Ecto schema module corresponding to the item
+  * `item_query` (function): A function that modifies the base query. This function should accept an Ecto.Queryable and return an Ecto.Queryable. It's used to apply additional query logic.
+  * `fields` (list): A list of atoms representing the fields to be selected and potentially preloaded.
+  """
+  def get(id, repo, schema, item_query, fields) do
+    query_record(id, repo, schema, item_query, fields, :one)
```

Comment:
I find that a bit difficult to read. Maybe write it like that?

```suggestion
    record_query(id, repo, schema, item_query, fields)
    |> Repo.one!()
  end

  @doc """
  Gets a database record with the given fields by the given id. Returns nil if no result was found.

  ## Parameters

  * `id`: The identifier for the specific item to be fetched.
  * `repo` (module): The repository module.
  * `schema`: The Ecto schema module corresponding to the item
  * `item_query` (function): A function that modifies the base query. This function should accept an Ecto.Queryable and return an Ecto.Queryable. It's used to apply additional query logic.
  * `fields` (list): A list of atoms representing the fields to be selected and potentially preloaded.
  """
  def get(id, repo, schema, item_query, fields) do
    record_query(id, repo, schema, item_query, fields)
    |> Repo.one()
```